snuba/datasets/table_storage.py

```diff
@@ -18,6 +18,7 @@ class KafkaTopicSpec:
     topic_name: str
     partitions_number: int
     replication_factor: int = 1
+    configuration: Mapping[str, str] = field(default_factory=dict)
```

Comment:
Is it our choice not to provide partitions_number and replication_factor as part of the configuration dictionary but as separate attributes or is it required by the confluent api ?

Reply:
Kafka thing. The partition count and replication factor are set and updated separately from the remainder of the topic configuration in both the admin client and the command line interface.